lib/trino-parquet/src/main/java/io/trino/parquet/reader/PageReader.java

```diff
@@ -56,12 +64,14 @@ public DataPage readPage()
         }
         DataPage compressedPage = compressedPages.remove(0);
         try {
+            long firstRowIndex = getFirstRowIndex(pageIndex++, offsetIndex);
```

Comment:
For clarity, move the `++` to a separate line. It's easier to reason about the code when there are no side-effects.